qiskit_experiments/library/characterization/t1_analysis.py

```diff
@@ -12,202 +12,68 @@
-            AnalysisError: if the analysis fails.
+    def _generate_fit_guesses(
+        self, user_opt: curve.FitOptions
+    ) -> Union[curve.FitOptions, List[curve.FitOptions]]:
+        """Apply conversion factor to tau."""
+        conversion_factor = self._experiment_options()["conversion_factor"]
+
+        if user_opt.p0["tau"] is not None:
+            user_opt.p0["tau"] *= conversion_factor
+
+        return super()._generate_fit_guesses(user_opt)
+
+    def _evaluate_quality(self, fit_data: curve.FitData) -> Union[str, None]:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - a reduced chi-squared lower than three
+            - absolute amp is within [0.9, 1.1]
+            - base is less than 0.1
+            - amp error is less than 0.1
+            - tau error is less than its value
+            - base error is less than 0.1
         """
-        data = experiment_data.data()
-        unit = data[0]["metadata"]["unit"]
-        conversion_factor = data[0]["metadata"].get("dt_factor", None)
-        qubit = data[0]["metadata"]["qubit"]
-
-        if conversion_factor is None:
-            conversion_factor = 1 if unit == "s" else apply_prefix(1, unit)
-
-        xdata, ydata, sigma = process_curve_data(data, lambda datum: level2_probability(datum, "1"))
-        xdata *= conversion_factor
-
-        if t1_guess is None:
-            t1_guess = np.mean(xdata)
-        else:
-            t1_guess = t1_guess * conversion_factor
-        if offset_guess is None:
-            offset_guess = ydata[-1]
-        if amplitude_guess is None:
-            amplitude_guess = ydata[0] - offset_guess
-
-        # Perform fit
-        def fit_fun(x, a, tau, c):
-            return a * np.exp(-x / tau) + c
-
-        init = {"a": amplitude_guess, "tau": t1_guess, "c": offset_guess}
-        fit_result = curve_fit(fit_fun, xdata, ydata, init, sigma=sigma)
-        fit_result = dataclasses.asdict(fit_result)
-        fit_result["circuit_unit"] = unit
-        if unit == "dt":
-            fit_result["dt"] = conversion_factor
-
-        # Construct analysis result
-        name = "T1"
-        unit = "s"
-        value = FitVal(fit_result["popt"][1], fit_result["popt_err"][1], unit="s")
-        chisq = fit_result["reduced_chisq"]
-        quality = self._fit_quality(
-            fit_result["popt"], fit_result["popt_err"], fit_result["reduced_chisq"]
-        )
-        analysis_results = [
-            AnalysisResultData(
-                name,
-                value,
-                chisq=chisq,
-                quality=quality,
-                extra=fit_result,
-            )
+        amp = fit_data.fitval("amp")
+        tau = fit_data.fitval("tau")
+        base = fit_data.fitval("base")
+
+        criteria = [
+            fit_data.reduced_chisq < 3,
+            abs(amp.value - 1.0) < 0.1,
+            abs(base.value) < 0.1,
+            amp.stderr is None or amp.stderr < 0.1,
```

Comment:
Same question as before about the stderr being None, can it happen in a successful fit